### PR TITLE
Stationbound Nerf

### PR DESCRIPTION
**Pull request by Geevies**
* A synthetic bolting an airlock is no longer instant, taking FIVE seconds to finish. The airlock announces when this process begins.
* A synthetic electrifying an airlock is no longer instant, taking FIVE seconds to finish. The airlock announces when this process begins.
* The above does not apply to antagonist synthetics.
* Forcefields can no longer be machine overloaded.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by SenpaiShadow** [comment]
Not ok with removing the hotkey thing at all. It was a convenient feature, being restricted to the menu alone is ass.

**Comment by Geevies** [comment]
Opening and closing the door can still be done with hotkeys. Bolting and electrifying cannot. I should have specified.

**Comment by Sindorman** [comment]
what is the menu that AI has to use to bolt doors? Is it a list of all doors?

**Comment by NonQueueingMatt** [comment]
@Sindorman Clicking a door opens a menu of actions you can do to that door (bolt/bolt lights/remove power...)

**Comment by skull132** [comment]
Removing hotkeys or other QoL features like that is an ass of an bandaid fix. Please don't actually do those.

**Comment by Sindorman** [comment]
Agreed with skull

**Comment by skull132** [comment]
What is "Remote electrification" and how does it differ from normal electrification? Unless you mean that the AI can no longer electrify doors, and the only way to do that is via wires.

**Comment by Geevies** [comment]
> 
> 
> What is "Remote electrification" and how does it differ from normal electrification? Unless you mean that the AI can no longer electrify doors, and the only way to do that is via wires.

Yah.

**Comment by Kaedwuff** [comment]
Bad faith update. About the only thing I like here if the bolt delay and announcement, but malfunctioning AI should have a way to get around it.

**Comment by Kaedwuff** [comment]
There still needs to be a way for malf AI to get around this, or you have effectively made them helpless.  This should be to prevent shitty AI behavior, not cripple antagonists.

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -847,14 +847,22 @@ About the new airlock wires panel:
...
+				to_chat(usr, "The door bolts will drop in three seconds.")
+				src.visible_message("\icon[src.icon] <b>[src]</b> announces, <span class='notice'>\"Bolts set to drop in THREE SECONDS.\"</span>")
+				addtimer(CALLBACK(src, .proc/lock), 30)
+			else if(!activate)
+				to_chat(usr, "The door bolts will raise in three seconds.")
+				src.visible_message("\icon[src.icon] <b>[src]</b> announces, <span class='notice'>\"Bolts set to raise in THREE SECONDS.\"</span>")
+				addtimer(CALLBACK(src, .proc/unlock), 30)
 		if("electrify_temporary")
-			electrify(30 * activate, 1)
+			to_chat(usr, "The door will electrify in five seconds.")
```
Comment (NonQueueingMatt) [inline code/game/machinery/doors/airlock.dm]: Span classes bruh

File: code/game/machinery/doors/airlock.dm [hunk 2]
```diff
@@ -847,14 +847,22 @@ About the new airlock wires panel:
 		if("bolts")
 			if(src.isWireCut(AIRLOCK_WIRE_DOOR_BOLTS))
 				to_chat(usr, "The door bolt control wire is cut - Door bolts permanently dropped.")
-			else if(activate && src.lock())
-				to_chat(usr, "The door bolts have been dropped.")
-			else if(!activate && src.unlock())
-				to_chat(usr, "The door bolts have been raised.")
+			else if(activate)
+				to_chat(usr, "The door bolts will drop in three seconds.")
```
Comment (NonQueueingMatt) [inline code/game/machinery/doors/airlock.dm]: Span classes bruh

File: code/game/machinery/doors/airlock.dm [hunk 3]
```diff
@@ -847,14 +847,22 @@ About the new airlock wires panel:
...
-			else if(activate && src.lock())
-				to_chat(usr, "The door bolts have been dropped.")
-			else if(!activate && src.unlock())
-				to_chat(usr, "The door bolts have been raised.")
+			else if(activate)
+				to_chat(usr, "The door bolts will drop in three seconds.")
+				src.visible_message("\icon[src.icon] <b>[src]</b> announces, <span class='notice'>\"Bolts set to drop in THREE SECONDS.\"</span>")
+				addtimer(CALLBACK(src, .proc/lock), 30)
+			else if(!activate)
+				to_chat(usr, "The door bolts will raise in three seconds.")
```
Comment (NonQueueingMatt) [inline code/game/machinery/doors/airlock.dm]: Span classes bruh

File: code/game/machinery/doors/airlock.dm [hunk 4]
```diff
@@ -847,14 +847,22 @@ About the new airlock wires panel:
...
+				src.visible_message("\icon[src.icon] <b>[src]</b> announces, <span class='notice'>\"Bolts set to raise in THREE SECONDS.\"</span>")
+				addtimer(CALLBACK(src, .proc/unlock), 30)
 		if("electrify_temporary")
-			electrify(30 * activate, 1)
+			to_chat(usr, "The door will electrify in five seconds.")
+			src.visible_message("\icon[src.icon] <b>[src]</b> announces, <span class='notice'>\"Airlock set to electrify in FIVE SECONDS.\"</span>")
+			addtimer(CALLBACK(src, .proc/electrify, 30 * activate, 1), 50)
 		if("electrify_permanently")
-			electrify(-1 * activate, 1)
+			to_chat(usr, "The door will electrify in five seconds.")
```
Comment (NonQueueingMatt) [inline code/game/machinery/doors/airlock.dm]: Span classes bruh

File: code/game/machinery/doors/airlock.dm (hunk 2 above)
Comment (Geevies) [inline code/game/machinery/doors/airlock.dm]:
```suggestion
				to_chat(usr, span("notice", "The door bolts will drop in three seconds."))
```

File: code/game/machinery/doors/airlock.dm (hunk 3 above)
Comment (Geevies) [inline code/game/machinery/doors/airlock.dm]:
```suggestion
				to_chat(usr, span("notice", "The door bolts will raise in three seconds."))
```

File: code/game/machinery/doors/airlock.dm (hunk 1 above)
Comment (Geevies) [inline code/game/machinery/doors/airlock.dm]:
```suggestion
			to_chat(usr, span("notice", "The door will electrify in five seconds."))
```

File: code/game/machinery/doors/airlock.dm (hunk 4 above)
Comment (Geevies) [inline code/game/machinery/doors/airlock.dm]:
```suggestion
			to_chat(usr, span("notice", "The door will electrify in five seconds."))
```

File: html/changelogs/geeves - aiNerf.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes:
+  - rscadd: "Remotely bolting an airlock is no longer instant, taking THREE seconds to finish. The airlock announces when this process begins."
+  - rscadd: "Remotely electrifying an airlock is no longer instant, taking THREE seconds to finish. The airlock announces when this process begins."
```
Comment (Arrow768) [inline html/changelogs/geeves - aiNerf.yml]: changelog sais 3 seconds

File: code/game/machinery/doors/airlock.dm [hunk 6]
```diff
@@ -847,14 +847,22 @@ About the new airlock wires panel:
...
+				addtimer(CALLBACK(src, .proc/unlock), 30)
 		if("electrify_temporary")
-			electrify(30 * activate, 1)
+			to_chat(usr, span("notice", "The door will electrify in five seconds."))
+			src.visible_message("\icon[src.icon] <b>[src]</b> announces, <span class='notice'>\"Airlock set to electrify in FIVE SECONDS.\"</span>")
+			addtimer(CALLBACK(src, .proc/electrify, 30 * activate, 1), 50)
 		if("electrify_permanently")
-			electrify(-1 * activate, 1)
+			to_chat(usr, span("notice", "The door will electrify in five seconds."))
+			src.visible_message("\icon[src.icon] <b>[src]</b> announces, <span class='notice'>\"Airlock set to electrify in FIVE SECONDS.\"</span>")
```
Comment (Arrow768) [inline code/game/machinery/doors/airlock.dm]: code sais 5 seconds

**Comment by Geevies** [comment]
Can't be arsed to fix this right now, in the middle of exams. Feel free to close this if I haven't touched it by the 23th, Nov.

**Comment by Geevies** [comment]
Reopening. Let's see the merge conflicts.

**Comment by Chada-1** [comment]
This works too but I think we'd actually just be wholly, 100% better off just completely removing the ability to electrify and bolt (and maintaining their ability to unelectrify and unbolt).

**Comment by Geevies** [comment]
yeah im working on it

**Comment by Chada-1** [comment]
> yeah im working on it

O 👀

**Comment by Geevies** [comment]
ok so basically, im monki

**Pull request closed without merging**